whylog/front/__init__.py

```diff
@@ -0,0 +1,37 @@
+from whylog.client import WhylogClient
+from whylog.teacher import WhylogTeacher, WhylogChecker
+
+
+class WhylogFront(object):
+    """
+    This class is the only one that will be used by plugins for text editors
+    implementing Whylog.
+    It provide all funtionalities of whylog.
+    """
+
+    def __init__(self, **opening_params):
+        self._whylog_client = WhylogClient(**opening_params)
+        self._whylog_teacher = WhylogTeacher(**opening_params)
+
+    def get_causes(self, front_input):
+        """
+        Function returning result of the investigation of the front input
+        """
+        return self._whylog_client.get_causes(front_input)
+
+    def get_tree(self, front_input):
+        return self._whylog_client.get_tree(front_input)
+
+    def new_teach_object(self):
+        """
+        This function return object that can teach new dependence for whylog
+        """
+        return self._whylog_teacher.new_teach_object()
+
+    @staticmethod
+    def check_teach_object(teach_object, front_input):
+        check_client = WhylogChecker(teach_object)
+        return check_client.get_causes(front_input)
+
+    def add_teach_object(self, teach_object):
+        self._whylog_teacher.add_dependence(teach_object)
```

Comment:
Apparently the caller of WhylogFront will have the teach object already and will just pass it here. It can be only added, not updated or removed, and the only way to get a teach object is to call `new_teach_object()` which takes no parameters. I am not saying this is a good or bad situation, I am just asking if this is what you meant to do.

Could you explain what interface will teach object expose and why is it called a teach object?